Add autolabels example

srtena opened the pull request:
## Summary
  This PR adds a new example demonstrating the `autoLabels` feature of VectorTileLayer available since deck.gl 9.2: https://deck.gl/docs/api-reference/carto/vector-tile-layer#autolabels

  The example includes two demonstrations accessible via a dropdown selector:
  - **Polygons**: US counties with automatic labels at centroids, styled by population
  - **Lines**: Bristol UK cycle network with labels along routes, styled by status

  ## Specs
  - Dropdown to switch between polygon and line demonstrations
  - Smooth view state transitions using `FlyToInterpolator` when switching between examples
  - Uses deck.gl 9.2 and @carto/api-client 0.5.21

  ## Screenshots
  ### Polygons Example
  <img width="1800" height="1039" alt="Screenshot 2025-10-29 at 18 39 52" src="https://github.com/user-attachments/assets/b736d626-ce67-46ab-997e-15d840ab33b9" />

  ### Lines Example
<img width="1800" height="1036" alt="Screenshot 2025-11-04 at 13 47 01" src="https://github.com/user-attachments/assets/0a3a3472-5ab9-4e28-a7c1-0a3df6d25bb3" />


  ## Test plan
  - ✅ Build successful
  - ✅ Dev server runs without errors
  - ✅ Both polygon and line examples display correctly
  - ✅ Labels have appropriate contrast against different backgrounds
  - ✅ Smooth transitions between examples
  - ✅ Consistent with other examples in structure and documentation

srtena committed:
Add autolabels example

This example demonstrates the autoLabels feature of VectorTileLayer, which automatically generates labels for polygon features at their centroids. The example displays US counties with population-based styling and automatic label generation with collision detection.

srtena committed:
Improve text readability with better line-height

felixpalmer commented on autolabels/README.md:
```diff
@@ -0,0 +1,27 @@
+## Example: Auto Labels
+
+This example demonstrates the `autoLabels` feature of VectorTileLayer, which automatically generates labels for polygon features at their centroids.
```
```suggestion
This example demonstrates the [autoLabels feature of VectorTileLayer](https://deck.gl/docs/api-reference/carto/vector-tile-layer#autolabels), which automatically generates labels for polygon features at their centroids.
```

felixpalmer commented on autolabels/index.html:
```diff
@@ -0,0 +1,39 @@
+      <div id="story-card">
+        <p class="overline">✨👀 You're viewing</p>
+        <h2>Auto Labels for Polygons</h2>
+        <p>This example demonstrates the <code>autoLabels</code> feature of <a href="https://deck.gl/docs/api-reference/carto/vector-tile-layer">VectorTileLayer</a>, which automatically generates labels for polygon features at their centroids.</p>
```
```suggestion
        <p>This example demonstrates the <code>autoLabels</code> feature of <a href="https://deck.gl/docs/api-reference/carto/vector-tile-layer#autolabels">VectorTileLayer</a>, which automatically generates labels for polygon features at their centroids.</p>
```

felixpalmer reviewed (approved):
Look great!

srtena committed:
Update autolabels/index.html

Co-authored-by: felixpalmer <felixpalmer@gmail.com>

srtena committed:
Update autolabels/README.md

Co-authored-by: felixpalmer <felixpalmer@gmail.com>

srtena committed:
Add line example with dropdown selector

- Add dropdown to switch between polygons and lines examples
- Implement Bristol UK cycle network example with colorCategories
- Add smooth FlyToInterpolator transitions between examples
- Update descriptions dynamically based on selected example
- Both examples showcase autoLabels for different geometry types

srtena committed:
Improve line label contrast with inverted colors
